app/services/post_alerter.rb

```diff
@@ -639,12 +639,17 @@ def notify_pm_users(post, reply_to_user, notified)
   end
 
   def group_notifying_via_smtp(post)
-    return nil if !SiteSetting.enable_smtp || post.post_type != Post.types[:regular]
+    return if !SiteSetting.enable_smtp || post.post_type != Post.types[:regular]
+    return if post.topic.allowed_groups.none?
```

Comment:
technically in a weird world `post.topic` can be nil...

Reply:
This is already handled further up https://github.com/discourse/discourse/blob/10914a4ffde995098d6c313b9110e3ba6fe3d579/app/services/post_alerter.rb#L602-L608